Solutions/Corvus.Azure.Storage.Tenancy/Corvus/Azure/Storage/Tenancy/ITenantCloudTableFactory.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="ITenantCloudTableFactory.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Corvus.Azure.Storage.Tenancy
+{
+    using System.Threading.Tasks;
+    using Corvus.Tenancy;
+    using Microsoft.Azure.Cosmos.Table;
+    using Microsoft.Azure.Storage.Blob;
+
+    /// <summary>
+    /// A factory for a <see cref="CloudTable"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// You use this type to get an instance of an <see cref="CloudTable"/> for a specific
+    /// <see cref="ITenant"/>. It uses a KeyVault to get the storage account key for the tenant, and the
+    /// configuration comes from the tenant via <see cref="TableStorageTenantExtensions.AddTableStorageConfiguration(System.Collections.Generic.IEnumerable{System.Collections.Generic.KeyValuePair{string, object}}, TableStorageTableDefinition, TableStorageConfiguration)"/>.
+    /// </para>
+    /// <para>
+    /// To configure a simple single-tenanted solution, which can ultimately be extended to multitenancy, the easiest route is to configure a configuration-based account key
+    /// provider and configuration for your repositories.
+    /// </para>
+    /// <para>
+    /// First, add the table factory and the configuration account key provider in your table configuration (assuming you have added a standard ConfigurationRoot to your solution).
+    /// </para>
+    /// <code>
+    /// serviceCollection.AddTenantCloudTableFactory();
+    /// serviceCollection.AddTenantConfigurationAccountKeyProvider();
+    /// </code>
+    /// <para>
+    /// <code>
+    /// TenantCloudTableFactory factory;
+    ///
+    /// var repository = await factory.GetTableForTenantAsync(tenantProvider.Root, new TableStorageContainerDefinition("sometable"));
```

Comment:
Splitting hairs probably, but `repository` seems an odd choice here. Why not... `table`?